core-rust/state-manager/src/receipt.rs

```diff
@@ -129,7 +144,7 @@ pub struct LedgerTransactionReceipt {
     /// Its omitted details may be found in `LocalTransactionExecution::outcome`.
     pub outcome: LedgerTransactionOutcome,
     /// The substate changes resulting from the transaction.
-    pub substate_changes: Vec<SubstateChange>,
+    pub substate_changes: BySubstate<ChangeAction>,
```

Comment:
_(note from author)_

This is a not-strictly-necessary-but-very-related refactor.

Initially I introduced `BySubstate` for substates' `SystemStructure`s, since passing around the Engine's `IndexMap<NodeId, IndexMap<PartitionNumber, IndexMap<SubstateKey, SubstateSystemStructure>>>` was inconvenient.

Then I noticed that it is actually a better way of storing the substates themselves too.